subprojects/core/build.gradle.kts

```diff
@@ -77,6 +77,7 @@ dependencies {
     api(libs.jsr305)
     api(libs.nativePlatform)
 
+    implementation(project(":core-api-jvm"))
```

Comment:
I think this is where things are tricky.  I think we don't want core to depend on the new pieces.  Is this only needed for downstream consumers of core or does something in core compile against something in core-api-jvm?

Reply:
This is needed because several classes in core: (`JavaEcosystemAttributesDescriber`, `JavaEcosystemSupport`, and `DefaultScriptClassPathResolver`) compile against `TargetJvmVersion` and `TargetJvmEnvironment`.

It seems to make sense to move those `Target` types to a JVM-specific platform.  Unfortunately moving out the types that use them (into a `core-jvm`, for instance), doesn't look like it would be so simple, as they are tied together with the script handling classes.

I think detangling the API classes is a good first step, even if that leaves `core` in pretty much the same state, just with several `core-api-x` deps instead of a single one.

I added a note to the dependency.

Reply:
I think I agree with Sterling here. I would like to see if we can solve this without needing this dependency.